src/pytest_recording/plugin.py

```diff
@@ -94,17 +94,25 @@ def block_network(request: SubRequest, record_mode: str, vcr_markers: List[Mark]
     marker = request.node.get_closest_marker(name="block_network")
     if marker is not None:
         validate_block_network_mark(marker)
+    config = request.getfixturevalue("vcr_config")
```

Comment:
This change makes the `vcr_config` fixture eagerly evaluated for every test. Generally, I tend to avoid evaluating user-defined fixtures until they are needed. Maybe some `allowed_hosts` fixture would work as well - it could be evaluated in the conditional branch below.

Reply:
Though, it could be done separately